cmd/bosun/web/static/partials/graph.html

```diff
@@ -83,54 +83,114 @@
+							</div>
 						</div>
-					</div>
-					<div class="form-group" ng-show="t.derivative == 'counter'">
-						<label class="col-sm-4 control-label" tooltip title="A positive integer representing the maximum value for the counter.">Counter Max</label>
-						<div class="col-sm-8">
-							<input type="number" class="form-control" ng-model="t.rateOptions.counterMax">
+						<div class="form-group" ng-show="t.derivative == 'counter'">
+							<label class="col-sm-4 control-label" tooltip title="A positive integer representing the maximum value for the counter.">Max</label>
+							<div class="col-sm-8">
+								<input type="number" class="form-control" ng-model="t.rateOptions.counterMax">
+							</div>
 						</div>
-					</div>
-				</form>
-				<form class="form-horizontal col-sm-4">
-					<div class="form-group" ng-repeat="k in sorted_tagks[$index]">
-						<label class="col-sm-4 control-label" tooltip title="Tag values for the tag. Can have multiple seperate by | which will return a time series for each one. Can also be * which returns a time series for each possible value.">{{k}} ({{tagvs[$parent.$index][k].length}})</label>
-						<div class="col-sm-8">
-							<input type="text" class="form-control" ng-model="t.tags[k]" ng-options="tv for tv in tagvs[$parent.$index][k]" bs-typeahead min-length="0" limit="100" ng-disabled="!t.metric_tags[k]">
+					</form>
+					<form class="form-horizontal col-sm-8">
+						<div class="form-group" ng-repeat="k in sorted_tagks[$index]">
+							<label class="col-sm-2 control-label" tooltip title="Enter in values to filter tags. The first filter (function and value) will act as a group by function - each match will corespond to a series. The second filter for the tag will filter the results of the series; instead of generating a new series the filtered series will be aggregated.">{{k}} ({{tagvs[$parent.$index][k].length}})</label>
+							<div class="col-sm-2">
+								<select title="Function to be used for a groupby filter." class="form-control" ng-model="t.gbFilters[k].type" ng-options="f for f in filters"></select>
+							</div>
+							<div class="col-sm-3">
+								<input type="text" class="form-control" ng-model="t.gbFilters[k].filter" ng-options="tv for tv in tagvs[$parent.$index][k]" bs-typeahead min-length="0" limit="100" ng-disabled="!t.metric_tags[k]">
+							</div>
+							<div class="col-sm-2">
+								<select title="Function to be used for a non-groupby filter." class="form-control" ng-model="t.nGbFilters[k].type" ng-options="f for f in filters"></select>
+							</div>
+							<div class="col-sm-3">
+								<input type="text" class="form-control" ng-model="t.nGbFilters[k].filter" ng-options="tv for tv in tagvs[$parent.$index][k]" bs-typeahead min-length="0" limit="100" ng-disabled="!t.metric_tags[k]">
+							</div>
 						</div>
-					</div>
-				</form>
+					</form>
+				</div>
+
+				<div ng-if="! filterSupport">
```

Comment:
I like this duplication now, just because we can delete it when we depricate 2.1. Maybe a comment somewhere reminding us to make changes to both blocks though.